main.cc

```diff
@@ -962,7 +962,7 @@ int main(int ac, char** av) {
             //FIXME: discarded future
             (void)api::set_server_cache(ctx);
             startlog.info("Waiting for gossip to settle before accepting client requests...");
-            gms::get_local_gossiper().wait_for_gossip_to_settle().get();
+            gms::get_local_gossiper().wait_for_gossip_to_settle(stop_signal.as_local_abort_source()).get();
```

Comment:
In  storage_service class, we have 
 
   abort_source& _abort_source

member. Why not do this to gossiper class as well so we can avoid passing abort_source to all the calls.

Reply:
The idea is to let the caller of the operation supply an `abort_source`, so that they control when the operation is cancelled. Initially, I did it in the way you described, but changed after comments from @tgrabiec.

Reply:
There is only one caller that is storage_service.cc. So every abort source passed to gossiper is from storage_service which in turn from the object from main.cc. What's the point to make storage_service to pass abort source?

Reply:
@asias In general, having one abort source per component is inflexible.

First, we may want to interrupt operations not only when the user hits ^C. If an operation on gossiper is part of a background operation started by storage_service, we will want to interrupt that operation when storage_service::stop() is called. When we fail to boot with an exception we will trigger a cascade of stops on each component (according to the depenency chain), without the main `abort_source` being triggered. 

Another problem is that, in general, interrupting all operations from all components at once may not be correct due to dependencies. If component A depends on B, then A needs to be stopped before B is stopped. If you interrupt B before stopping A, then A may not be able to stop gracefully. B::stop() should interrupt background operations managed by B, only after A stopped itself.

Those are theoretical problems though for now, so I committed the patch.